src/components/app.jsx

```diff
@@ -30,23 +31,35 @@ export default class App extends React.Component {
       case 'review-cards':
         return <ReviewCards/>;
       case 'view-cards':
-        return <ViewCards/>;
+        return <ViewCards cardList={this.state.cards} onClick={this.getCards}/>;
       default:
         return null;
     }
   }
 
+  componentDidMount() {
+    this.getCards();
+  }
+
+  getCards() {
+    const savedCardList = JSON.parse(localStorage.getItem('flash-cards'));
+    this.setState({
+      cards: savedCardList
+    });
+  }
+
   saveCards() {
     const cardsList = JSON.stringify(this.state.cards);
     localStorage.setItem('flash-cards', cardsList);
+    this.getCards();
   }
 
   addCard(card) {
     const cardsList = [...this.state.cards];
     cardsList.push(card);
     this.setState({
       cards: cardsList
-    }, this.saveCards);
+    }, this.saveCards());
```

Comment:
Be careful not to call this method too soon. You should not invoke the function here. Another way would be

```js
}, () => this.saveCards());
```

we don't want the `saveCards` method to be executed until _after_ `setState` completes.